python/mlcroissant/mlcroissant/_src/beam.py

```diff
@@ -0,0 +1,70 @@
+"""Beam module."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import typing
+from typing import Any
+
+from etils import epath
+
+from mlcroissant._src.datasets import Dataset
+
+if typing.TYPE_CHECKING:
+    import apache_beam as beam
+
+
+def ReadFromCroissant(
+    *,
+    pipeline: beam.Pipeline,
+    jsonld: epath.PathLike | Mapping[str, Any],
+    record_set: str,
+    mapping: Mapping[str, epath.PathLike] | None = None,
+):
+    """Returns an Apache Beam reader to generate the dataset using e.g. Spark.
+
+    Example of usage:
+
+    ```python
+    import apache_beam as beam
+    from apache_beam.options import pipeline_options
+    import mlcroissant as mlc
+
+    jsonld = "https://huggingface.co/api/datasets/ylecun/mnist/croissant"
+
+    options = pipeline_options.PipelineOptions()
+    with beam.Pipeline(options=options) as pipeline:
+        mlc.ReadFromCroissant(
+            pipeline=pipeline,
+            jsonld=jsonld,
+            record_set="default",
+        )
+    ```
+
+    Only streamable datasets can be used with Beam. A streamable dataset is a dataset
+    that can be generated by a linear sequence of operations - without joins for
+    example. This is the case for Hugging Face datasets. If there are branches, we'd
+    need a more complex Beam pipeline.
```

Comment:
So IIUC we will change this in future work to work with non-streamable datasets (e.g. future versions of HF croissants) -- I feel if this is right, this would deserve a mention here, or a link to an issue?

Reply:
Done.